service/cachemdw/caching_middleware.go

```diff
@@ -0,0 +1,60 @@
+package cachemdw
+
+import (
+	"net/http"
+
+	"github.com/kava-labs/kava-proxy-service/decode"
+)
+
+// CachingMiddleware returns kava-proxy-service compatible middleware which works in the following way:
+// - tries to get decoded request from context (previous middleware should set it)
+// - checks few conditions:
+//   - if request isn't already cached
+//   - if request is cacheable
+//   - if response is present in context
+//
+// - if all above is true - caches the response
+// - calls next middleware
+func (c *ServiceCache) CachingMiddleware(
+	next http.Handler,
+) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		// if cache is not enabled - do nothing and forward to next middleware
+		if !c.cacheEnabled {
+			next.ServeHTTP(w, r)
```

Comment:
trace log that we're skipping caching since it's disabled

Reply:
done